bundler/spec/quality_spec.rb

```diff
@@ -253,9 +253,62 @@ def check_for_specific_pronouns(filename)
     expect(lib_code).to eq(spec_code)
   end
 
+  it "documents all cli command options in their associated man pages" do
+    commands = normalize_commands_and_options(Bundler::CLI)
+    cli_and_man_pages_in_sync!(commands)
+
+    Bundler::CLI.subcommand_classes.each do |_, klass|
+      subcommands = normalize_commands_and_options(klass)
+
+      cli_and_man_pages_in_sync!(subcommands)
+    end
+  end
+
   private
 
   def each_line(filename, &block)
     File.readlines(filename, encoding: "UTF-8").each_with_index(&block)
   end
+
+  def normalize_commands_and_options(command_class)
+    commands = {}
+
+    command_class.commands.each do |command_name, command|
+      next if command.is_a?(Bundler::Thor::HiddenCommand)
+
+      key = command.ancestor_name || command_name
+      commands[key] ||= []
+      # Verify that all subcommands are documented in the main command's man page.
+      commands[key] << command_name unless command_class == Bundler::CLI
+
+      command.options.each do |name, option|
+        commands[key] << name.to_s.tr("_", "-") unless option.banner.to_s.include?("(deprecated)")
```

Comment:
I wonder if checking specifically for the flag name, `--all` for example, will be less likely to have false positives. `option.switch_name` would give you that. I guess that won't work for subcommands, but we could split the check and check subcommands separately.

Reply:
Maybe deprecated flags would be documented too, just mentioning that they are deprecated. But good to skip them for now regardless to get this passing more quickly.

Reply:
> I wonder if checking specifically for the flag name, --all

Yeah great suggestion, done 👍.

--- 

> Maybe deprecated flags should be documented too,

I made the change to check for deprecated flags. However, the `bundle plugin install --local-git` was expressly removed as part of deprecating it in #7048. I have added it to the allowlist, but I'm happy to add the doc back if you prefer.

Reply:
I don't have a strong opinion, but I think we can add it back to the documentation. Previously I believe it was only present in the command synopsis. If we add it back, I'd also explicitly mention that it's a deprecated alias of `--git`.